pythonforandroid/graph.py

```diff
@@ -99,17 +132,129 @@ def find_order(graph):
                 bset.discard(result)
 
 
-def get_recipe_order_and_bootstrap(ctx, names, bs=None):
-    recipes_to_load = set(names)
+def obvious_conflict_checker(ctx, name_tuples, blacklist=[]):
+    """ This is a pre-flight check function that will completely ignore
+        recipe order or choosing an actual value in any of the multiple
+        choice tuples/dependencies, and just do a very basic obvious
+        conflict check.
+    """
+    deps_were_added_by = dict()
+    deps = set()
+
+    # Add dependencies for all recipes:
+    to_be_added = [(name_tuple, None) for name_tuple in name_tuples]
+    while len(to_be_added) > 0:
+        current_to_be_added = list(to_be_added)
+        to_be_added = []
+        for (added_tuple, adding_recipe) in current_to_be_added:
+            assert(type(added_tuple) == tuple)
```

Comment:
Minor: why would we `assert` that? why not covering the case in a test then?
Also the `assert` statement is better without parenthesis because you can fall into a common pitfall using them

Reply:
It is just an input data check. I put it there because I was stupid and put in wrong input data, I probably forgot to remove it (but it's not per se harmful)

Basically `obvious_conflict_checker` doesn't accept the mixed string/tuple lists as the overall graph function but only already tuple transformed lists and at some point I forgot. This assert hits if you screw that up, which never happens normally since nobody is supposed to call the function like that (it's only called from the graph code that already does this cleanup previously)

So similarly as above, what test case would you have in mind? If I remove the assert it's also going to fail, so the test wouldn't really verify the assert. The assert is really just there to catch it earlier

Reply:
The test case I have in mind is simply the test that shows this assert raising

Reply:
hm fair enough, I suppose I could check for `AssertionError`. I'll put something in :+1:

Reply:
added a test! :heart: hope it's okay if I resolve this, otherwise please slap me

Reply:
Yes you wrote it as I imagined thanks